File: Igtampe.CDBFS.Common/AccessRecord.cs
namespace Igtampe.CDBFS.Common {

    public enum Access {
        READ = 0,
        WRITE = 1,
        OWNER = 2,
    };

    public class AccessRecord {
        public int Id { get; set; }

        /// <summary>Specific username this record relates to. If it's null, it means this is the entry for public access</summary>
        public string? Username { get; set; } = null;

        public int DriveId { get; set; } = 0;

        /// <summary>Allows a user to Read all the files </summary>
        public Access Access { get; set; } = Access.READ;

    }
}

File: igtampe.cdbfs.frontend/src/model/AccessRecord.ts
import CdbfsDrive from "./CdbfsDrive";

export enum AccessLevel {
    READ = 0,
    WRITE = 1,
    OWNER = 2
}

export default class AccessRecord {
    public constructor(
        public id: number,
        public username: string,
        public driveId: number,
        public access: AccessLevel,
        public drive?: CdbfsDrive
    ) { }
}